### PR TITLE
Implement fuzzy search

### PR DESCRIPTION
**Pull request by Bearfinn**

### CODE REVIEW

File: src/app/SearchSection.tsx [hunk 1]
```diff
@@ -57,17 +58,21 @@ const SearchSection = () => {
   const { isDesktop } = useCustomMediaQuery();
   const [data, setData] = React.useState<Shop[]>([]);
   const [isLoading, setLoading] = React.useState<boolean>(true);
+  const [db, setDb] = React.useState<MiniSearch>();
```
Comment (hspotlight) [inline src/app/SearchSection.tsx]: I don't see the usage of db and setDb. Maybe we can remove it (including import line)

File: src/app/SearchSection.tsx [hunk 2]
```diff
@@ -57,17 +58,21 @@ const SearchSection = () => {
...
+    searchService.initialize();
     searchService.setData(result.data);
+
     setLoading(false);
   };
 
-  const searchTextOnType = (givenText: string) => {
-    const result = searchService.searchText(givenText);
-    setData(result);
+  const searchTextOnType = async (givenText: string) => {
```
Comment (hspotlight) [inline src/app/SearchSection.tsx]: Do we need to make this function async?

Comment (Bearfinn) [reply src/app/SearchSection.tsx]: Updated!

File: src/app/SearchSection.tsx (hunk 1 above)
Comment (Bearfinn) [reply src/app/SearchSection.tsx]: Updated!

**Review by hspotlight — APPROVED** [review]
LGTM kub. thank you for contribution kub. :)

**Pull request merged**